src/components/Timelog/WeeklySummaries.jsx

```diff
@@ -48,31 +64,39 @@ const WeeklySummaries = ({ userProfile }) => {
     const newEditedSummaries = [...editedSummaries];
     newEditedSummaries[index] = userProfile.weeklySummaries[index]?.summary || '';
     setEditedSummaries(newEditedSummaries);
-    
+  
     // Toggle off editing mode
     toggleEdit(index);
-  };
 
+  };
+  
   const handleSave = async (index) => {
+    setLoadingHandleSave(index);
     // Save the edited summary content and toggle off editing mode
     const editedSummary = editedSummaries[index];
-    // Check if the edited summary is not blank and contains at least 50 words
-    const wordCount = editedSummary.split(/\s+/).filter(Boolean).length;
+
     if (editedSummary.trim() !== '' && wordCount >= 50) {
       const updatedUserProfile = {
         ...userProfile,
         weeklySummaries: userProfile.weeklySummaries.map((item, i) =>
-          i === index ? { ...item, summary: editedSummary } : item
+        i === index ? { ...item, summary: editedSummary } : item
         )
       };
-  
-    await dispatch(updateUserProfile(userProfile._id, updatedUserProfile));
+      
+    // This code updates the summary.  
+    await dispatch(updateUserProfile(userProfile));
+    
+    // This code saves edited weekly summaries in MongoDB.
+    await dispatch(updateWeeklySummaries(userProfile._id, updatedUserProfile));
     await dispatch(getUserProfile(userProfile._id));
-      // Toggle off editing mode
-      toggleEdit(index);
-    } else {
-      // Invalid summary, show an error message or handle it as needed
-      alert('Please enter a valid summary with at least 50 words.');
+    await setLoadingHandleSave(null);
```

Comment:
Delete this line. Sorry I quoted the wrong code snippet in my other comment

Reply:
Hi [Abi-Liu](https://github.com/Abi-Liu).  I removed the setLoadingHandleSave from the handleSave function.

![photo 3](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/73541319/5c9949c3-83e5-44b7-b09b-47a11b586a79)

Reply:
It's still there. Underneath `await dispatch(getUserProfile(userProfile._id));`